Jenkinsfile

```diff
@@ -168,10 +170,77 @@ pipeline {
                 }
                 stage('Build on Ubuntu 18.04') {
                     agent {
-                        label 'docker_runner'
+                        dockerfile {
+                            filename 'Dockerfile.ubuntu.18.04'
+                            label 'docker_runner'
+                            additionalBuildArgs '--build-arg UID=$(id -u)' +
+                              ' --build-arg CACHEBUST=' +
+                               currentBuild.startTimeInMillis
+                        }
                     }
                     steps {
-                        echo "Building on Ubuntu is not implemented for the moment"
+                        sh '''rm -rf artifacts/ubuntu18.04/
+                              mkdir -p artifacts/ubuntu18.04/
+                              : "${DEBEMAIL:="$env.DAOS_EMAIL"}"
+                              : "${DEBFULLNAME:="$env.DAOS_FULLNAME"}"
+                              export DEBEMAIL
+                              export DEBFULLNAME
+                              make debs'''
+                    }
+                    post {
+                        success {
+                            sh '''ln -v \
+                                   _topdir/BUILD/*{.build,.changes,.deb,.dsc,.gz,.xz} \
+                                   artifacts/ubuntu18.04/
+                                  pushd artifacts/ubuntu18.04/
+                                    dpkg-scanpackages . /dev/null | \
+                                      gzip -9c > Packages.gz
+                                  popd'''
+                            archiveArtifacts artifacts: 'artifacts/ubuntu18.04/**'
+                        }
+                        failure {
+                            sh script: "cat _topdir/BUILD/*.build",
+                               returnStatus: true
+                            archiveArtifacts artifacts: 'artifacts/ubuntu18.04/**'
+                        }
+                    }
+                }
+                stage('Build on Ubuntu 18.10') {
```

Comment:
Do we need to build on 18.10?

Reply:
But wouldn't we need to build Ubuntu 18.10 packages for everything else to be able to do this?  I must not be understanding.

Reply:
A package built for 18.04 should be able to be installed on an 18.10 system and possibly an 19.x system.
So unless a package has a dependency on Ubuntu 18.10, we do not need to build an 18.10 variant for a comparison test.

Reply:
> A package built for 18.04 should be able to be installed on an 18.10 system and possibly an 19.x system.

That's a crapshoot.  There are no guarantees.  Any number of shared libraries can change between releases making a package not work (for various values of "not work" that can include outright not function, or appear to function but do strange things, etc.) on a foreign release.

> So unless a package has a dependency on Ubuntu 18.10, we do not need to build an 18.10 variant for a comparison test.

I think that kind of mixing and matching can introduce it's own variables, so trying to use it to eliminate some other variables is suspect at best.

Anyway, I'm not that adamantly against building this one outlier 18.10 component.  I'm just not convinced what use it will be.